jumpcloud-ADMU/Powershell/Start-Migration.ps1

```diff
@@ -1368,6 +1387,137 @@ function Set-ADMUScheduledTask {
+                foreach ($ext in $exts) {
+                    $indivExtension = $ext.PSChildName
+                    $progId = (Get-ItemProperty "$($pathRoot)\$indivExtension\UserChoice" -ErrorAction SilentlyContinue).ProgId
+                    $manifestList += [PSCustomObject]@{
+                        extension  = $indivExtension
+                        programId = $progId
+                    }
+                }
+                Write-ToLog "Found $($manifestList.count) Associations"
+            } else {
+                Write-ToLog "No File Type Association path found"
+            }
+
+
+        }
+        catch {
+            Write-ToLog "Error getting file type associations$($_.Exception.Message)"
+        }
+    }
+    end {
+        if ($manifestList) {
+            Write-ToLog -Message "Writing fta_manifest.csv to $($profilePath)\AppData\Local\JumpCloudADMU\fta_manifest.csv"
+            $manifestList | Export-CSV ($profilePath + '\AppData\Local\JumpCloudADMU\fta_manifest.csv') -Force
+        } else {
+            Write-ToLog -Message "No File Type Associations found"
+        }
+    }
+}
+
+# Get user protocol associations
+
+function Get-ProtocolTypeAssociation{
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true, HelpMessage = 'The SID of the user to capture file type associations')]
+        [System.String]
+        $UserSid,
+        [Parameter(Mandatory = $true, HelpMessage = 'The profile path of the new user to store the file type associations')]
+        [System.String]
+        $profilePath
+    )
+    begin {
+        Write-ToLog "Getting Protocol Type Associations for userSID: $($UserSid)_admu"
+    }
+    process {
+        $manifestList = @()
+        try {
+            $pathRoot = "HKEY_USERS:\$($UserSid)_admu\Software\Microsoft\Windows\Shell\Associations\UrlAssociations\"
+            if (Test-Path $pathRoot) {
+                Get-ChildItem $pathRoot* |
+                ForEach-Object {
+
+                    $progId = (Get-ItemProperty "$($_.PSParentPath)\$($_.PSChildName)\UserChoice" -ErrorAction SilentlyContinue).ProgId
+                    if ($progId) {
+                        $manifestList += [PSCustomObject]@{
+                            extension  = $_.PSChildName
+                            programId = $progId
+                        }
+                    }
+                }
+                Write-ToLog "Found $($manifestList.count) Associations"
+            } else {
+                Write-ToLog "No Protocol Association path found"
+            }
+        }
+        catch {
+            # Write the error to the log
+            Write-ToLog "Error getting protocol associations$($_.Exception.Message)"
+        }
+    }
+    end {
+        if ($manifestList) {
+            Write-ToLog -Message "Writing protocol_manifest.csv to $($profilePath)\AppData\Local\JumpCloudADMU\protocol_manifest.csv"
+
+            # Export list to JumpCloudADMU folder
```

Comment:
I'd want us to break this out of this function. Any Get function should return something, it wasn't clear to me when I was first reading the code where this was being written. So I'd like to see the Get function, return a list and write out the list to a file within `Start-Migration` function

Reply:
Fixed it. The get-functions returns list now